### PR TITLE
Add support for custom objective functions in UniversalFallback.

### LINKED ISSUE #1085: Feature request: Support custom objective functions in UniversalFallback
I can create a custom `MOI.AbstractFunction` and add support to it through bridges. The custom function can then be included in constraints, before attaching a bridged optimizer, because the `UniversalFallback` supports any constraints. It would be nice if this also worked for the objective function.

Currently, if I add a custom objective function that is not supported by the `MOIU.Model` it ends up in the `modattr` dict of `UniversalFallback`. When JuMP queries the objective function it first gets the `ObjectiveFunctionType` from the `MOIU.Model` so that ultimately a `zero(ScalarAffineFunction{Float64})` will be returned, even if I have added support for my custom function in JuMP and MOI through bridges.

`UniversalFallback` could handle custom objectives that are not supported by `MOIU.Model` in the same way that it currently handles custom constraints.

### PR DESCRIPTION
**Pull request by martinbiel**
Quick prototype that resolves #1085 

Tests are missing.

### CODE REVIEW

File: src/Utilities/universalfallback.jl [hunk 1]
```diff
@@ -54,11 +57,12 @@ function Base.show(io::IO, U::UniversalFallback)
 end
 
 function MOI.is_empty(uf::UniversalFallback)
-    return MOI.is_empty(uf.model) && isempty(uf.constraints) &&
+    return MOI.is_empty(uf.model) && isnothing(uf.objective) && isempty(uf.constraints) &&
```
Comment (blegat) [inline src/Utilities/universalfallback.jl:60]: Don't use `isnothing` as it is not available in Julia v1.0

File: src/Utilities/universalfallback.jl [hunk 2]
```diff
@@ -2,17 +2,18 @@
     UniversalFallback
 
 The `UniversalFallback` can be applied on a [`MathOptInterface.ModelLike`](@ref)
-`model` to create the model `UniversalFallback(model)` supporting *any*
-constaint and attribute. This allows to have a specialized implementation in
-`model` for performance critical constraints and attributes while still
+`model` to create the model `UniversalFallback(model)` supporting *any* objective,
```
Comment (blegat) [inline src/Utilities/universalfallback.jl:5]: Technically, the objective is an attribute. The objective attribute needs a special handling by `UniversalFallback` but it's still an attribute.

File: src/Utilities/universalfallback.jl [hunk 3]
```diff
@@ -156,6 +160,9 @@ function MOI.delete(uf::UniversalFallback, vi::VI)
     for d in values(uf.varattr)
         delete!(d, vi)
     end
+    if !(uf.objective === nothing)
```
Comment (blegat) [inline src/Utilities/universalfallback.jl:163]: You can use `!==`

File: src/Utilities/universalfallback.jl [hunk 4]
```diff
@@ -268,6 +281,52 @@ function MOI.get(uf::UniversalFallback, listattr::MOI.ListOfConstraintAttributes
     return list
 end
 
+# Objective
+MOI.supports(uf::UniversalFallback, ::MOI.ObjectiveSense) = true
```
Comment (blegat) [inline src/Utilities/universalfallback.jl:285]: This is already covered by
https://github.com/JuliaOpt/MathOptInterface.jl/blob/74d74433c783a980fa118ffbda1fbe9f5f0f1d8c/src/Utilities/universalfallback.jl#L339

File: src/Utilities/universalfallback.jl [hunk 5]
```diff
@@ -268,6 +281,52 @@ function MOI.get(uf::UniversalFallback, listattr::MOI.ListOfConstraintAttributes
...
+end
+function MOI.get(uf::UniversalFallback,
+                 attr::MOI.ObjectiveFunction{F})::F where F
+    if uf.objective === nothing
+        return MOI.get(uf.model, attr)
+    else
+        return uf.objective
+    end
+end
+MOI.supports(uf::UniversalFallback, ::MOI.ObjectiveFunction{F}) where F <: MOI.AbstractScalarFunction = true
```
Comment (blegat) [inline src/Utilities/universalfallback.jl:309]: This is already covered by
https://github.com/JuliaOpt/MathOptInterface.jl/blob/74d74433c783a980fa118ffbda1fbe9f5f0f1d8c/src/Utilities/universalfallback.jl#L339

File: src/Utilities/universalfallback.jl [hunk 6]
```diff
@@ -268,6 +281,52 @@ function MOI.get(uf::UniversalFallback, listattr::MOI.ListOfConstraintAttributes
...
+function MOI.set(uf::UniversalFallback,
+                 attr::MOI.ObjectiveFunction,
+                 func::MOI.AbstractScalarFunction)
+    if MOI.supports(uf.model, attr)
+        MOI.set(uf.model, attr, func)
+        uf.objective = nothing
+    else
+        uf.objective = copy(func)
+        uf.model.objectiveset = false
```
Comment (blegat) [inline src/Utilities/universalfallback.jl:318]: `uf.model` is not necessarily an `MOIU.AbstractModel`. Here, you can do:
```julia
sense = MOI.get(uf.model, MOI.ObjectiveSense())
MOI.set(uf.model, MOI.ObjectiveSense(), MOI.FEASIBILITY_SENSE) # clear objective function if any was set
MOI.set(uf.model, MOI.ObjectiveSense(), sense)
```

**Comment by blegat** [comment]
Implementation looks good. Just missing tests

**Comment by martinbiel** [comment]
I added some simple tests to `universalfallback.jl`, but more importantly I believe 
```
function MOI.supports(
    ::ModelForUniversalFallback{T},
    ::Type{MOI.ObjectiveFunction{MOI.ScalarAffineFunction{T}}}) where T
    return false
end
```
should give a lot of coverage from the unit tests. I also rebased the branch on master.

**Pull request merged**